prov/mrail/src/mrail_ep.c

```diff
@@ -382,6 +382,9 @@ mrail_send_common(struct fid_ep *ep_fid, const struct iovec *iov, void **desc,
 		FI_WARN(&mrail_prov, FI_LOG_EP_DATA,
 			"Unable to fi_sendmsg on rail: %" PRIu32 "\n", i);
 		goto err2;
+	} else if (!(flags & FI_COMPLETION)) {
+		/* Inject won't generate a completion, so we count it here */
```

Comment:
The comment can be dropped. Only absence of FI_COMPLETION would not generate completion. Passing only FI_INJECT flag can still generate completion.

Reply:
Gotcha, I'll update the comment to say that no completion was requested. Does that work for you?